api/hypershift/v1beta1/gcp.go

```diff
@@ -1,5 +1,42 @@
 package v1beta1
 
+// GCPResourceReference represents a reference to a GCP resource by name.
+// Follows GCP naming patterns (name-based APIs, not ID-based like AWS).
+// See https://google.aip.dev/122 for GCP resource name standards.
+type GCPResourceReference struct {
+	// name is the name of the GCP resource
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=`^[a-z]([a-z0-9]*(-[a-z0-9]+)*)?$`
+	Name string `json:"name"`
+}
+
+// GCPEndpointAccessType defines the endpoint access type for GCP clusters.
+// Equivalent to AWS EndpointAccessType but adapted for GCP networking model.
+type GCPEndpointAccessType string
+
+const (
+	// GCPEndpointAccessPublicAndPrivate endpoint access allows public API server access and
+	// private node communication with the control plane via PSC.
+	GCPEndpointAccessPublicAndPrivate GCPEndpointAccessType = "PublicAndPrivate"
+
+	// GCPEndpointAccessPrivate endpoint access allows only private API server access and private
+	// node communication with the control plane via PSC.
+	GCPEndpointAccessPrivate GCPEndpointAccessType = "Private"
+)
+
+// GCPNetworkConfigCustomer specifies customer VPC configuration for GCP clusters and PSC endpoint creation.
+type GCPNetworkConfigCustomer struct {
+	// network is the customer's VPC network name
+	// +required
+	Network GCPResourceReference `json:"network"`
+
+	// pscSubnet is the customer's subnet for PSC endpoints
+	// +required
+	PSCSubnet GCPResourceReference `json:"pscSubnet"`
```

Comment:
we should avoid using abbreviations in API field names a doc. please replace `PSC` with the full qualified name.
can't this field name just be `subnet`?